File: sdkwork-ai-sdk-typescript/src/api/responses.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { CancelResponsePostResponse, CreateResponseRequest, CreateResponseResponse, DeleteResponseResponse, GetResponseResponse, ListResponseInputItemsResponse } from '../types';


export class ResponsesApi2 {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Cancel response */
  async cancelResponse(response_id: string | number): Promise<CancelResponsePostResponse> {
    return this.client.post<CancelResponsePostResponse>(aiApiPath(`/v1/responses/${response_id}/cancel`));
  }

/** Create response */
  async createResponse(body: CreateResponseRequest): Promise<CreateResponseResponse> {
    return this.client.post<CreateResponseResponse>(aiApiPath(`/v1/responses`), body);
  }

/** List response input items */
  async listResponseInputItems(response_id: string | number): Promise<ListResponseInputItemsResponse> {
    return this.client.get<ListResponseInputItemsResponse>(aiApiPath(`/v1/responses/${response_id}/input_items`));
  }

/** Get response */
  async getResponse(response_id: string | number): Promise<GetResponseResponse> {
    return this.client.get<GetResponseResponse>(aiApiPath(`/v1/responses/${response_id}`));
  }

/** Delete response */
  async deleteResponse(response_id: string | number): Promise<DeleteResponseResponse> {
    return this.client.delete<DeleteResponseResponse>(aiApiPath(`/v1/responses/${response_id}`));
  }
}

export function createResponsesApi2(client: HttpClient): ResponsesApi2 {
  return new ResponsesApi2(client);
}

File: sdkwork-ai-sdk-flutter/lib/src/api/responses.dart
import '../http/client.dart';
import '../models.dart';

class ResponsesApi {
  final HttpClient _client;
  
  ResponsesApi(this._client);

  /// Cancel response
  Future<CancelResponseResponse?> cancelResponse(String response_id) async {
    final response = await _client.post(ApiPaths.aiPath('/responses/${response_id}/cancel'));
    return response is CancelResponseResponse ? response : null;
  }

  /// Create response
  Future<CreateResponseResponse?> createResponse(CreateResponseRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/responses'), body: body, contentType: 'application/json');
    return response is CreateResponseResponse ? response : null;
  }

  /// List response input items
  Future<ListResponseInputItemsResponse?> listResponseInputItems(String response_id) async {
    final response = await _client.get(ApiPaths.aiPath('/responses/${response_id}/input_items'));
    return response is ListResponseInputItemsResponse ? response : null;
  }

  /// Get response
  Future<GetResponseResponse?> getResponse(String response_id) async {
    final response = await _client.get(ApiPaths.aiPath('/responses/${response_id}'));
    return response is GetResponseResponse ? response : null;
  }

  /// Delete response
  Future<DeleteResponseResponse?> deleteResponse(String response_id) async {
    final response = await _client.delete(ApiPaths.aiPath('/responses/${response_id}'));
    return response is DeleteResponseResponse ? response : null;
  }
}
